sdk/endpoint/composite.go

```diff
@@ -19,82 +19,71 @@ import (
 	"context"
 
 	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/sirupsen/logrus"
 	"google.golang.org/grpc"
 
 	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/connection"
 	"github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/local/networkservice"
+	"github.com/networkservicemesh/networkservicemesh/utils/typeutils"
 )
 
 // InitContext is the context passed to the Init function of the endpoint
 type InitContext struct {
 	GrpcServer *grpc.Server
 }
 
-// ChainedEndpoint is the basic endpoint composition interface
-type ChainedEndpoint interface {
-	networkservice.NetworkServiceServer
-	Name() string
-	Init(context *InitContext) error
-	GetNext() ChainedEndpoint
-	GetOpaque(interface{}) interface{}
```

Comment:
Need to update sdk documentation here: https://github.com/networkservicemesh/networkservicemesh/blob/master/sdk/README.md#writing-a-chainedendpoint